src/NuGet.Jobs.RegistrationComparer/Normalizers.cs

```diff
@@ -0,0 +1,266 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text.RegularExpressions;
+using Newtonsoft.Json.Linq;
+using NuGet.Versioning;
+
+namespace NuGet.Jobs.RegistrationComparer
+{
+    public class Normalizers
+    {
+        private readonly static HashSet<string> PackageIdsWithDotDotDependencies = new HashSet<string>(StringComparer.OrdinalIgnoreCase)
+        {
+            "angularjs.TypeScript.DefinitelyTyped",
+            "devextreme.TypeScript.DefinitelyTyped",
+        };
+
+        private static readonly IReadOnlyList<KeyValuePair<Func<string, bool>, Func<JToken, bool, ComparisonContext, string>>> DefaultScalarNormalizers =
+            new List<KeyValuePair<Func<string, bool>, Func<JToken, bool, ComparisonContext, string>>>
```

Comment:
The types are a bit complex. Do you think it'd be easier to understand them if we added a few abstractions? Maybe:

* Func<string, bool>` could be a delegate named `PathMatcher`
* `Func<JToken, bool, ComparisonContext, string>` could be a delegate named `Normalize`
* `KeyValuePair<...>` could be a private type named `Normalizer`

Reply:
(I don't feel too strongly about this, feel free to ignore if you feel the abstractions would just make things more confusing)

Reply:
Great idea. These shapes morphed over time so I didn't think it was right at first but now they are stable.

Reply:
Done.